Document stategic merge patch. (#3823)
prometheus-operator/prometheus-operator: Documentation/user-guides/strategic-merge-patch.md

# Strategic merge patch

When users need to apply a specific configuration to containers that we do not support or do not currently exist, merge patch can be used.
This document describes how to overwrite the container configuration generated by the operator by merging patches.

## How the "strategic merge patch" works
	

> Operator support `containers` field in `PrometheusSpec` and `AlertmanagerSpec` configuration.

Reviewer: ```suggestion
The operator supports `containers` field in `PrometheusSpec`, `AlertmanagerSpec` and `ThanosRulerSpec` configuration.
```

This field allows injecting additional containers, and the existing configuration can be overwritten by sharing the same container name.

> ### Merge patch Prometheus example

Reviewer: ```suggestion
### Merge patch example of Prometheus
```

> The following is an example on how to overwrite Prometheus container `readinessProbe`  `FailureThreshold` configuration.

Reviewer: ```suggestion
The following manifest overwrites the `failureThreshold` value of the readiness probe for the Prometheus container.
```

```
apiVersion: monitoring.coreos.com/v1
kind: Prometheus
metadata:
  labels:
    prometheus: self
  name: self
  namespace: default
spec:
  containers:
  - name: prometheus

>     readinessProbe

Reviewer: ```suggestion
    readinessProbe:
```

      failureThreshold: 500
```

> ### Merge patch alertmanager example

Reviewer: ```suggestion
### Merge patch example for Alertmanager
```

> Overwrite alertmanager container readinessProbe  `FailureThreshold` configuration.

Reviewer: ```suggestion
The following manifest overwrites the `failureThreshold` values of the readiness and liveness probes for the Alertmanager container.
```

```
apiVersion: monitoring.coreos.com/v1
kind: Alertmanager
metadata:
  labels:
    alertmanager: main
  name: main
  namespace: monitoring
spec:
  containers:
  - name: alertmanager
    livenessProbe:
      failureThreshold: 5

>     readinessProbe

Reviewer: ```suggestion
    readinessProbe:
```

# Strategic merge patch

> Some users find some unreasonable container configurations,they want to have modifications on it.

Reviewer 2: ```suggestion
When users need to apply a specific configuration to containers that we do not support or do not currently exist, merge patch can be used.
```

This document describes how to overwrite the container configuration generated by the operator by merging patches.

## How the "strategic merge patch" works
	
Operator support `containers` field in `PrometheusSpec` and `AlertmanagerSpec` configuration.

> This field allows injecting additional containers,and the existing configuration can be overwritten by sharing the same conatiner name.

Reviewer 2: ```suggestion
This field allows injecting additional containers, and the existing configuration can be overwritten by sharing the same container name.
```

> ### Merge patch prometheus example

Reviewer 2: ```suggestion
### Merge patch Prometheus example
```

> Overwrite prometheus container readinessProbe  `FailureThreshold` configuration.

Reviewer 2: ```suggestion
The following is an example on how to overwrite Prometheus container `readinessProbe`  `FailureThreshold` configuration.
```
